### PR TITLE
Feature/farm 483 ratings firebase repo

### PR DESCRIPTION
**Pull request by LeeMatthewHiggins**
[FARM-483 - ratings firebase repo]

#### 📲 What

Implemented the firebase ratings repo
Refactored the ratings system to use simple maps, to allow for new factors to be added in the future. 
Refactored recommendation engine to do user plot factors lookup internally to make the setup easier. Now gets weights and factors in one call (which is how its implemented in firebase database)
		
#### 🛠 How
		
The factors and weights are now fetched from the server.
Currently only using the crop names to match data, using the lookup collection will be in next PR. 
The plot and user repos still use the mock,  so nothing will be persisted after app close.

#### 👀 See
		
not much to see, just ratings will now have a % that's not zero.
Currently using mock plot info with only 3 factors, so %´s will be low.
		
#### 🕵️‍♂️ How to test

open add and view recommendations screen. (add to plot)

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/model/repositories/ratingEngine/implementation/MockRatingEngineRepository.dart [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
+      return MapEntry(subject, RatingInfo(harryWeights[subject], value));
+    });
+    return Future.value(inputFactors);
   }
 
   @override
-  Stream<List<FactorEntity>> observeWeights() {
-    return observeList();
+  Future<Map<String, RatingInfo>> observeRatingInfo() {
+    // TODO: implement observeRatingInfo
```
Comment (albertmiro) [inline lib/model/repositories/ratingEngine/implementation/MockRatingEngineRepository.dart]: Just to be sure, that we didn't miss it, that would go with another PR?

**Review by albertmiro — DISMISSED** [review]

File: lib/model/repositories/ratingEngine/implementation/MockRatingEngineRepository.dart (hunk 1 above)
Comment (LeeMatthewHiggins) [reply lib/model/repositories/ratingEngine/implementation/MockRatingEngineRepository.dart]: In the mock it will never change, but I should implement to be complete. 👍

Comment (LeeMatthewHiggins) [reply lib/model/repositories/ratingEngine/implementation/MockRatingEngineRepository.dart]: done

**Review by albertmiro — APPROVED** [review]
LGTM!
